scipy/_lib/doccer.py

```diff
@@ -112,9 +126,9 @@ def inherit_docstring_from(cls):
     >>> b = Bar()
     >>> b.func.__doc__
     'Do something useful.\n        Do it fast.\n        '
-
     """
-    def _doc(func):
+
+    def _doc(func: Callable[P, R]) -> Callable[P, R]:
```

Comment:
The same reasoning as for `Decorator` applies here; this could be made even better by binding the entire `Callable` as `def _doc(func: _F) -> _F`. 

And if wou really want to dot the "i"'s here, you could consider making `func` positional-only